### PR TITLE
Big interface patch

### PR DESCRIPTION
**Pull request by Julius-Syvis**
1) jei mes serialization error delete bin/data folder, ten bus outdated
2) daug refactor changes
3) super interface added (so abstract)

### CODE REVIEW

File: EmotionCalculator/EmotionCalculator/FormsUI/BaseForm.cs [hunk 1]
```diff
@@ -111,71 +117,95 @@ private void OpenFileButton_Click(object sender, EventArgs e)
 
         private void CameraStartButton_Click(object sender, EventArgs e)
         {
-            cam.Start();
+            cameraHandle.Start();
+
             camStartButton.Enabled = false;
+
+            submitWebCamButton.Enabled = true;
```
Comment (Mantulis00) [inline EmotionCalculator/EmotionCalculator/FormsUI/BaseForm.cs:124]: puike

File: EmotionCalculator/EmotionCalculator/FormsUI/BaseForm.cs [hunk 2]
```diff
@@ -111,71 +117,95 @@ private void OpenFileButton_Click(object sender, EventArgs e)
...
         private async void SubmitWebCamButton_Click(object sender, EventArgs e)
         {
-            submitWebCamButton.Enabled = false;
+            DisableButtons();
+
             Image image = null;
-            string response;
-            if (cam.cameraIsRoling)
+
+            if (cameraHandle.cameraRunning)
```
Comment (Mantulis00) [inline EmotionCalculator/EmotionCalculator/FormsUI/BaseForm.cs:144]: galima nuimt, nes neįmanoma paspaust mygtuko jei kamera neįjungta ?

**Review by Mantulis00 — APPROVED** [review]
Galima būtų mergint, bet, manau, pirma reiktų CrashFix sumergint, kad nekonfliktintų.

**Comment by Mantulis00** [comment]
System.Runtime.Serialization.SerializationException: 'Unable to load type System.Collections.Generic.Dictionary`2[[System.Int32, mscorlib

**Comment by Mantulis00** [comment]
> System.Runtime.Serialization.SerializationException: 'Unable to load type System.Collections.Generic.Dictionary`2[[System.Int32, mscorlib

https://stackoverflow.com/questions/14990980/system-runtime-serialization-serializationexception-unable-to-find-assembly-mya

**Comment by Julius-Syvis** [comment]
parašiau pirmam komentare, tiesiog ištrink bin/data ir viskas veiks, ten outdated serialized failai 👍

File: EmotionCalculator/EmotionCalculator/FormsUI/BaseForm.cs (hunk 2 above)
Comment (Julius-Syvis) [reply EmotionCalculator/EmotionCalculator/FormsUI/BaseForm.cs:144]: manau bonus protection netrukdo

**Pull request merged**